drivers/include/periph/pwm.h

```diff
@@ -18,84 +18,91 @@
 
-/* ignore file in case no PWM devices are defined */
-#if PWM_NUMOF
-
 /**
- * @brief Definition of available PWM devices
- *
- * To this point a maximum of four PWM device is available.
+ * @brief   Default PWM access macro
+ * @{
  */
-typedef enum {
-#if PWM_0_EN
-    PWM_0,              /*< 1st PWM device */
-#endif
-#if PWM_1_EN
-    PWM_1,              /*< 2nd PWM device */
+#ifndef PWM_DEV
+#define PWM_DEV(x)          (x)
 #endif
-#if PWM_2_EN
-    PWM_2,              /*< 3rd PWM device */
+/** @} */
+
+/**
+ * @brief  Default PWM undefined value
+ * @{
+ */
+#ifndef PWM_UNDEF
+#define PWM_UNDEF           (-1)
 #endif
-#if PWM_3_EN
-    PWM_3,              /*< 4th PWM device */
+/** @} */
+
+/**
+ * @brief   Default PWM type definition
+ * @{
+ */
+#ifndef HAVE_PWM_T
+typedef int pwm_t;
 #endif
-} pwm_t;
+/** @} */
 
 /**
- * @brief Definition of available PWM modes
+ * @brief   Default PWM mode definition
+ * @{
  */
+#ifndef HAVE_PWM_MODE_T
 typedef enum {
     PWM_LEFT,           /*< use left aligned PWM */
     PWM_RIGHT,          /*< use right aligned PWM */
     PWM_CENTER          /*< use center aligned PWM */
 } pwm_mode_t;
+#endif
+/** @} */
 
 /**
  * @brief Initialize a PWM device
  *
- * The PWM module is based on virtual PWM devices, which can have one or more channels.
- * The PWM devices can be configured to run with a given frequency and resolution, which
- * are always identical for the complete device, hence for every channel on a device.
+ * The PWM module is based on virtual PWM devices, which can have one or more
+ * channels. The PWM devices can be configured to run with a given frequency and
+ * resolution, which are always identical for the complete device, hence for
+ * every channel on a device.
  *
- * The desired frequency and resolution may not be possible on a given device when chosen
- * too large. In this case the PWM driver will always keep the resolution and decrease the
- * frequency if needed. To verify the correct settings compare the returned value which
- * is the actually set frequency.
+ * The desired frequency and resolution may not be possible on a given device
+ * when chosen too large. In this case the PWM driver will always keep the
+ * resolution and decrease the frequency if needed. To verify the correct
+ * settings compare the returned value which is the actually set frequency.
  *
  * @param[in] dev           PWM channel to initialize
  * @param[in] mode          PWM mode, left, right or center aligned
- * @param[in] frequency     the PWM frequency in Hz
- * @param[in] resolution    the PWM resolution
+ * @param[in] freq          PWM frequency in Hz
+ * @param[in] res           PWM resolution
  *
- * @return                  Actual PWM frequency on success
- * @return                  -1 on mode not applicable
- * @return                  -2 on frequency and resolution not applicable
+ * @return                  actual PWM frequency on success
+ * @return                  0 on error
```

Comment:
How do existing applications (their implementers) get aware that their error checking got invalid?